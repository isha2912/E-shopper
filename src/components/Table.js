import React from 'react';
import PropTypes from 'prop-types';
import './Cart.scss';

const Table = (props) => {
  const { cartItems, className } = props;
  return (
    <div className="tab">
      <table className={className}>
        <thead>
          <tr className={className}>
            <th> ITEM DESCRIPTION </th>
            <th> UNIT PRICE</th>
            <th> QUANTITY </th>
            <th> SUBTOTAL </th>
          </tr>
        </thead>
        <tbody>
          {cartItems.map((eachItem) => (
            <tr key={eachItem.id}>
              <td>{eachItem.name}</td>
              <td>{eachItem.price}</td>
              <td>{eachItem.quantity * eachItem.count}</td>
              <td>{eachItem.price * eachItem.count}</td>
            </tr>
          )) }
        </tbody>
      </table>
    </div>
  );
};

export default Table;

Table.propTypes = {
  cartItems: PropTypes.arrayOf(PropTypes.shape({
    image: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    count: PropTypes.number.isRequired,
    quantity: PropTypes.number.isRequired,
  })).isRequired,
  className: PropTypes.string.isRequired,
};
