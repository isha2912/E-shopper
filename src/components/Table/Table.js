/* eslint-disable react/prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import '../Cart/Cart.scss';

const Table = (props) => {
  const { cartItems, className } = props;
  return (
    <div className="tab">
      <table className={className}>
        <thead>
          <tr className={className}>
            <th> ITEM DESCRIPTION </th>
            <th> UNIT PRICE</th>
            <th> QUANTITY </th>
            <th> SUBTOTAL </th>
          </tr>
        </thead>
        <tbody>
          {Object.keys(cartItems).map((eachCategory) => (
            <React.Fragment key={eachCategory}>
              <tr>
                <td>{eachCategory}</td>
              </tr>

              {cartItems[eachCategory].map((eachItem) => (
                <tr key={eachItem.id}>
                  <td>{eachItem.name}</td>
                  <td>{eachItem.price}</td>
                  <td>{eachItem.quantity * eachItem.count}</td>
                  <td>{eachItem.price * eachItem.count}</td>
                </tr>
              ))}
            </React.Fragment>
          ))}

        </tbody>
      </table>
    </div>
  );
};

export default Table;

Table.propTypes = {
  cartItems: PropTypes.shape({
    category: PropTypes.arrayOf(
      PropTypes.shape({
        count: PropTypes.number.isRequired,
        stock: PropTypes.number.isRequired,
        price: PropTypes.number.isRequired,
        name: PropTypes.string.isRequired,
      }),
    ),
  }).isRequired,
  className: PropTypes.string.isRequired,
};
