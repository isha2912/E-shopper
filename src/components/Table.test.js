import React from 'react';
import { screen, render } from '@testing-library/react';
import Table from './Table';

describe('Table component', () => {
  const mockProp = {
    cartItems: {
      fruits: [{
        id: 1,
        name: 'apple',
        price: 10,
        count: 2,
        quantity: 2,
      }],
    },
    className: 'table',
  };
  afterEach(() => jest.clearAllMocks());
  test('Should display table components ', () => {
    const container = render(<Table {...mockProp} />);
    expect({ container }).toMatchSnapshot();
  });
  test('Should display product details', () => {
    render(<Table {...mockProp} />);
    screen.getByText('apple');
    screen.getByText('10');
    screen.getByText('4');
    screen.logTestingPlaygroundURL();
  });
});
