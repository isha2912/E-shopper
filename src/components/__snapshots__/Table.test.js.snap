// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component Should display table components  1`] = `
Object {
  "container": Object {
    "asFragment": [Function],
    "baseElement": <body>
      <div>
        <div
          class="tab"
        >
          <table
            class="table"
          >
            <thead>
              <tr
                class="table"
              >
                <th>
                   ITEM DESCRIPTION 
                </th>
                <th>
                   UNIT PRICE
                </th>
                <th>
                   QUANTITY 
                </th>
                <th>
                   SUBTOTAL 
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  fruits
                </td>
              </tr>
              <tr>
                <td>
                  apple
                </td>
                <td>
                  10
                </td>
                <td>
                  4
                </td>
                <td>
                  20
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </body>,
    "container": <div>
      <div
        class="tab"
      >
        <table
          class="table"
        >
          <thead>
            <tr
              class="table"
            >
              <th>
                 ITEM DESCRIPTION 
              </th>
              <th>
                 UNIT PRICE
              </th>
              <th>
                 QUANTITY 
              </th>
              <th>
                 SUBTOTAL 
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                fruits
              </td>
            </tr>
            <tr>
              <td>
                apple
              </td>
              <td>
                10
              </td>
              <td>
                4
              </td>
              <td>
                20
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>,
    "debug": [Function],
    "findAllByAltText": [Function],
    "findAllByDisplayValue": [Function],
    "findAllByLabelText": [Function],
    "findAllByPlaceholderText": [Function],
    "findAllByRole": [Function],
    "findAllByTestId": [Function],
    "findAllByText": [Function],
    "findAllByTitle": [Function],
    "findByAltText": [Function],
    "findByDisplayValue": [Function],
    "findByLabelText": [Function],
    "findByPlaceholderText": [Function],
    "findByRole": [Function],
    "findByTestId": [Function],
    "findByText": [Function],
    "findByTitle": [Function],
    "getAllByAltText": [Function],
    "getAllByDisplayValue": [Function],
    "getAllByLabelText": [Function],
    "getAllByPlaceholderText": [Function],
    "getAllByRole": [Function],
    "getAllByTestId": [Function],
    "getAllByText": [Function],
    "getAllByTitle": [Function],
    "getByAltText": [Function],
    "getByDisplayValue": [Function],
    "getByLabelText": [Function],
    "getByPlaceholderText": [Function],
    "getByRole": [Function],
    "getByTestId": [Function],
    "getByText": [Function],
    "getByTitle": [Function],
    "queryAllByAltText": [Function],
    "queryAllByDisplayValue": [Function],
    "queryAllByLabelText": [Function],
    "queryAllByPlaceholderText": [Function],
    "queryAllByRole": [Function],
    "queryAllByTestId": [Function],
    "queryAllByText": [Function],
    "queryAllByTitle": [Function],
    "queryByAltText": [Function],
    "queryByDisplayValue": [Function],
    "queryByLabelText": [Function],
    "queryByPlaceholderText": [Function],
    "queryByRole": [Function],
    "queryByTestId": [Function],
    "queryByText": [Function],
    "queryByTitle": [Function],
    "rerender": [Function],
    "unmount": [Function],
  },
}
`;
